src/lightning/fabric/utilities/init.py

```diff
@@ -0,0 +1,50 @@
+# Copyright The Lightning AI team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable, Dict, Optional, Sequence
+
+from lightning.fabric.utilities.imports import _TORCH_GREATER_EQUAL_1_13
+
+if _TORCH_GREATER_EQUAL_1_13:
+    from torch.overrides import TorchFunctionMode
+
+    # From https://lernapparat.de/faster-model-init by Thomas Viehmann
+    class _EmptyInit(TorchFunctionMode):
```

Comment:
These conditional classes do not always play nicely with our package creation, do you think we can have it inside and raise an exception in init?
also, from a user perspective (yes I see it shall be protected and used, but...) it could be confusing that in some runtime depending on PT, our package has this class or does not have

Reply:
I agree that it would be nicer to check on `__init__` of the class if torch>=1.13

Reply:
Okay, so what you are saying is that I should inherit from `object` on torch <= 1.13. I can try that